### ISSUE TITLE
Macros conflicts with "stat" when _FILE_OFFSET_BITS is defined

### ISSUE THREAD

**Issue by lazka** [msys2/MINGW-packages#10591]
When _FILE_OFFSET_BITS=64 is defined (ideally Windows code shouldn't depend on it, but here we are) mingw-w64 does this:

https://github.com/mingw-w64/mingw-w64/blob/898141aa957b99812f2f4d2cf89255798d578d54/mingw-w64-headers/crt/sys/stat.h#L279

Which wrongly replaces various things:

1) `QFileInfo::stat()` in https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/io/qfileinfo.h#n192
2) https://github.com/arvidn/libtorrent/blob/95afba30e9acfd8411076a0b05f7ca69bf33b9bb/include/libtorrent/peer_connection_interface.hpp#L47 -> https://gist.github.com/lazka/ebbd33274dc7a0f2521d3a639d96589f

* In libtorrent I worked around it by not defining _FILE_OFFSET_BITS on Windows (it didn't depend on it) https://github.com/msys2/MINGW-packages/blob/53b559893d92bc71767ddaa002c08e898307712c/mingw-w64-libtorrent-rasterbar/0001-no-offset-64.patch
* In qt6 it's not so easy since it's a public header

We should report this to mingw-w64, maybe they have an idea.

**Comment by MehdiChinoune** [comment]
For qt5 we fixed it by adding these lines
```CPP
#ifdef stat
#undef stat
#endif
```
to https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/io/qfileinfo.h

**Comment by lazka** [comment]
I posted this to the mailing list some hours ago, but my email is stuck in review I guess.

**Comment by MehdiChinoune** [comment]
https://sourceforge.net/p/mingw-w64/mailman/message/37595000

**Comment by oltolm** [comment]
I also had this problem when porting rpcs3. Can something be done about it?

**Comment by oltolm** [comment]
It seems that this has been fixed.

**Comment by lazka** [comment]
Indeed, in https://github.com/mingw-w64/mingw-w64/commit/07e34538d5a7d5e86ab47c7ba47becb32f7d827e it seems

**Comment by oltolm** [comment]
Should we close it then?

**Comment by lazka** [comment]
Yeah.

The only remaining patch we have there I think is https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-qt6-base/009-qfileinfo-undefine-mingw-stat.patch which we should try to remove.
